### ISSUE TITLE
self.* = tf.concat([self.*, more_paths.*], axis=2) no gradient

### ISSUE THREAD

**Issue by qiangdadi** [NVlabs/sionna#589]
[test-code.zip](https://github.com/user-attachments/files/17089105/test-code.zip)
I found that the gradients cannot be tracked after `tf.concat` during backpropagation for paths.merge() op.
OS : ubuntu22.04
cuda : v12.6
The record class is wrote by myself. And  It is convenient for me to record some variables. The codes list here.
`class Record:
    
    def __init__(self):
        
        self.data = {}

    def add(self, name, value):
        
        self.data[name] = value

    def get(self, name):
        
        return self.data.get(name, None)

    def clear(self):
        
        self.data.clear()`

`def merge(self, more_paths, path_type):

        dtype = self._scene.dtype

        more_vertices = more_paths.vertices
        more_objects = more_paths.objects
        more_types = more_paths.types

        # The paths to merge must have the same number of sources and targets
        assert more_paths.targets.shape[0] == self.targets.shape[0],\
            "Paths to merge must have same number of targets"
        assert more_paths.sources.shape[0] == self.sources.shape[0],\
            "Paths to merge must have same number of targets"

        # Pad the paths with the lowest depth
        padding = self.vertices.shape[0] - more_vertices.shape[0]
        if padding > 0:
            more_vertices = tf.pad(more_vertices,
                                   [[0,padding],[0,0],[0,0],[0,0],[0,0]],
                                   constant_values=tf.zeros((),
                                                            dtype.real_dtype))
            more_objects = tf.pad(more_objects,
                                  [[0,padding],[0,0],[0,0],[0,0]],
                                  constant_values=-1)
        elif padding < 0:
            padding = -padding
            self.vertices = tf.pad(self.vertices,
                                   [[0,padding],[0,0],[0,0],[0,0],[0,0]],
                            constant_values=tf.zeros((), dtype.real_dtype))
            self.objects = tf.pad(self.objects,
                                  [[0,padding],[0,0],[0,0],[0,0]],
                                  constant_values=-1)

        # Merge types
        if tf.rank(self.types) == 0:
            merged_types = tf.repeat(self.types, tf.shape(self.vertices)[3])
        else:
            merged_types = self.types
        if tf.rank(more_types) == 0:
            more_types = tf.repeat(more_types, tf.shape(more_vertices)[3])

        self.types = tf.concat([merged_types, more_types], axis=0)

        # Concatenate all
        self.a = tf.concat([self.a, more_paths.a], axis=2)
        self.tau = tf.concat([self.tau, more_paths.tau], axis=2)
        self.theta_t = tf.concat([self.theta_t, more_paths.theta_t], axis=2)
        # hxq marked
        print("trying restore computation graph")
        self._phi_t = tf.concat([self._phi_t, more_paths.phi_t], axis=2)
        
        self.theta_r = tf.concat([self.theta_r, more_paths.theta_r], axis=2)
        self.phi_r = tf.concat([self.phi_r, more_paths.phi_r], axis=2)
        self.mask = tf.concat([self.mask, more_paths.mask], axis=2)
        self.vertices = tf.concat([self.vertices, more_vertices], axis=3)
        self.objects = tf.concat([self.objects, more_objects], axis=3)
        self.doppler = tf.concat([self.doppler, more_paths.doppler], axis=2)

        # hxq marked
        self._scene.record.add(f"{path_type}_all_paths.phi_t", self.phi_t)

        return self`
I added some codes in SolverPaths.compute_fields()
` # hxq marked

self._scene.record.add(f"befor_merge_scat_all_paths.phi_t", all_paths.phi_t)

all_paths = all_paths.merge(scat_paths, "scat")`
The test script is based on "Learning of Material Properties via Gradient Descent" to build to test my view.
